packages/tasit-action/src/contract/Contract.test.js

```diff
@@ -128,7 +128,10 @@ describe("TasitAction.Contract", () => {
 
     describe("should trigger Contract error event", async () => {
       it("on action error", async () => {
-        const errorListener = sinon.fake();
+        const errorListener = sinon.fake(() => {
+          // Note: Forcing listener to be called once
```

Comment:
Is this a hack we want to undo at some point?

Reply:
Yes, that is a way that I've found to achieve determinism in some cases. We can remove the unsubscription and change `expect(errorListener.callCount).to.equal(1);` to `expect(errorListener.callCount).to.be.at.least(1);` if will be better.

Reply:
It seems to me it addresses the symptom but not the root cause. The tests are deterministic, but the thing they are testing is still behaving non-deterministically

Reply:
Yeah, I think I'd personally prefer the other suggestion you just mentioned. That way the expectation matches what would happen if you really used it in the wild. Thanks!

Reply:
I'm assuming I'll wait on this change to merge the PR

Reply:
Done.

Reply:
Thanks! Resolving this